org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/renderers/DBBuilder.java

```diff
@@ -0,0 +1,535 @@
+        "CopyrightLabel  nvarchar NULL,\r\n"+
+        "Json            nvarchar NOT NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+  private void makePropertiesTable(Connection con) throws SQLException {
+    Statement stmt = con.createStatement();
+    stmt.execute("CREATE TABLE Properties (\r\n"+
+        "Key          integer NOT NULL,\r\n"+
+        "ResourceKey  integer NOT NULL,\r\n"+
+        "Code         varchar NOT NULL,\r\n"+
+        "Uri          varchar NULL,\r\n"+
+        "Description  varchar NULL,\r\n"+
+        "Type         varchar NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+  private void makeConceptsTable(Connection con) throws SQLException {
+    Statement stmt = con.createStatement();
+    stmt.execute("CREATE TABLE Concepts (\r\n"+
+        "Key          integer NOT NULL,\r\n"+
+        "ResourceKey  integer NOT NULL,\r\n"+
+        "ParentKey    integer NULL,\r\n"+
+        "Code         varchar NULL,\r\n"+
+        "Display      varchar NULL,\r\n"+
+        "Definition   varchar NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+  private void makeConceptPropertiesTable(Connection con) throws SQLException {
+    Statement stmt = con.createStatement();
+    stmt.execute("CREATE TABLE ConceptProperties (\r\n"+
+        "Key          integer NOT NULL,\r\n"+
+        "ResourceKey  integer NOT NULL,\r\n"+
+        "ConceptKey   integer NOT NULL,\r\n"+
+        "PropertyKey  integer NULL,\r\n"+
+        "Code         varchar NULL,\r\n"+
+        "Value        varchar NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+  private void makeDesignationsTable(Connection con) throws SQLException {
+    Statement stmt = con.createStatement();
+    stmt.execute("CREATE TABLE Designations (\r\n"+
+        "Key          integer NOT NULL,\r\n"+
+        "ResourceKey  integer NOT NULL,\r\n"+
+        "ConceptKey   integer NOT NULL,\r\n"+
+        "UseSystem    varchar NOT NULL,\r\n"+
+        "UseCode      varchar NOT NULL,\r\n"+
+        "Lang         varchar NOT NULL,\r\n"+
+        "Value        text NOT NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+  private void makeMappingsTable(Connection con) throws SQLException {
+    Statement stmt = con.createStatement();
+    stmt.execute("CREATE TABLE ConceptMappings (\r\n"+
+        "Key           integer NOT NULL,\r\n"+
+        "ResourceKey   integer NOT NULL,\r\n"+
+        "SourceSystem  varchar NOT NULL,\r\n"+
+        "SourceVersion varchar NOT NULL,\r\n"+
+        "SourceCode    varchar NOT NULL,\r\n"+
+        "Relationship  varchar NOT NULL,\r\n"+
+        "TargetSystem  varchar NOT NULL,\r\n"+
+        "TargetVersion varchar NOT NULL,\r\n"+
+        "TargetCode    varchar NOT NULL,\r\n"+
+        "PRIMARY KEY (Key))\r\n");
+  }
+
+
+  public String processSQL(String sql) {
+    if (con == null) {
+      return "<span style=\"color: maroon\">Error processing SQL: SQL is not set up properly</span>";
+    }
+
+    sql = sql.trim();
```

Comment:
Null sql here could blow up. I'd suggest doing something like this in the catch block:

```java
 if (sql == null) {
        throw new IllegalArgumentException("Param sql cannot be null.");
      }
     
      sql = sql.trim();
```

Reply:
ok.

Reply:
I moved that into the try block. Feel free to revert that commit if it's improper, but I think it's safer to do it that way.

Reply:
no that's ok